### PR TITLE
Template sets codehausmojo, errorprone (was grpc, now split in order to reduce size) 

### PR DESCRIPTION
**Pull request by moljac**
### Describe your contribution

1. addded new bindings required as dependencies of other template sets (grpc, opencensus, perfmark)
  - `com.google.errorprone:error_prone_annotations`
  - `org.codehaus.mojo:animal-sniffer-annotations`
  
6. created template sets

## Updated/new (moved) artifacts

- com.google.errorprone:error_prone_annotations -  -> 2.4.0
- com.google.android:annotations -  -> 4.1.1.4
- org.codehaus.mojo:animal-sniffer-annotations -  -> 1.18

### Does this change any of the generated binding API's?

`net6-android` TFM added

### Artifacts updated

Checklist
- [x] Add any metadata needed to get PR to compile
- [x] Verify any new namespaces are properly renamed (utilities.cake)
- [x] Verify nuget/namespace spelling (utilities.cake)

### CODE REVIEW

**Review by jpobst — COMMENTED** [review]
Can we remove the unrelated changes in files:
- config.json
- GooglePlayServicesProject.cshtml
- source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml

**Comment by moljac** [comment]
> Can we remove the unrelated changes in files:
> 
> * config.json
> * GooglePlayServicesProject.cshtml
> * source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml

Will try to

**Comment by moljac** [comment]
> Can we remove the unrelated changes in files:
> 
> * config.json
> * GooglePlayServicesProject.cshtml
> * source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml

I think I removed all that I could

File: config.json [hunk 1]
```diff
@@ -1623,6 +1623,32 @@
...
+        "groupId": "com.google.errorprone",
+        "artifactId": "error_prone_annotations",
+        "version": "2.4.0",
+        "nugetVersion": "2.4.0",
+        "nugetId": "Xamarin.Google.ErrorProne.Annotations",
+        "dependencyOnly": false,
+        "templateSet": "errorprone"
+      },
+      {
+        "groupId": "com.google.android",
```
Comment (jpobst) [inline config.json:1637]: I assume this will need to be removed once #546 is merged.

File: templates/codehaus-mojo/External-Dependency-Info.txt [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+Do not translate or localize
+
+Xamarin Components for GoogleGson incorporates 
+third party material from the projects listed below. The original copyright 
+notice and the license under which Microsoft received such third party 
+material are set forth below.  Microsoft reserves all other rights not 
+expressly granted, whether by implication, estoppel or otherwise.
+
+########################################
+# GoogleGson
```
Comment (jpobst) [inline templates/codehaus-mojo/External-Dependency-Info.txt:11]: "GoogleGson"? Wrong license file?

**Review by jpobst — COMMENTED** [review]
Probably need to change the license file text, but other than that looks good.

File: templates/codehaus-mojo/External-Dependency-Info.txt (hunk 2 above)
Comment (moljac) [reply templates/codehaus-mojo/External-Dependency-Info.txt:11]: those are new bindings, so I forgot to change.

fixing

File: config.json (hunk 1 above)
Comment (moljac) [reply config.json:1637]: true

**Review by jpobst — APPROVED** [review]

**Pull request merged**